tests/testthat/test-runner.R

```diff
@@ -359,10 +359,87 @@ test_that("status missing ID", {
     status = "missing",
     version = NULL,
     output = NULL,
-    task_position = 0
+    queue = list()
   ))
 })
 
+test_that("check_timeout kills timed out reports", {
+  testthat::skip_on_cran()
+  skip_on_appveyor()
+  skip_on_windows()
+  skip_if_no_redis()
+  path <- orderly_prepare_orderly_example("demo")
+  runner <- orderly_runner(path, workers = 2)
+
+  key1 <- runner$submit_task_report("minimal", timeout = 0)
+  key2 <- runner$submit_task_report("slow10", timeout = 0)
+  ## Sleep for a bit to allow "minimal" report to post TASK_COMPLETE message
+  Sys.sleep(3)
+  msg <- capture_messages(killed <- runner$check_timeout())
+  task2 <- get_task_id_key(runner, key2)
+  expect_equal(killed, task2)
+  expect_equal(msg, sprintf("Successfully killed '%s', exceeded timeout of 0\n",
+                            task2))
+})
+
+test_that("check_timeout doesn't kill reports with long timeout", {
+  testthat::skip_on_cran()
+  skip_on_appveyor()
+  skip_on_windows()
+  skip_if_no_redis()
+  path <- orderly_prepare_orderly_example("demo")
+  runner <- orderly_runner(path, workers = 2)
+
+  key1 <- runner$submit_task_report("minimal", timeout = 20)
+  key2 <- runner$submit_task_report("slow10", timeout = 20)
+  Sys.sleep(1) ## Sleep so both reports get started
```

Comment:
this looks potentially flakey, especially if minimal might complete within time.

in other contexts I've rigged up things to poll status to wait until both things are started, which is probably preferable to the sleep as CI systems vary in speed. But minimal can be completed quite quickly so might escape kill by finishing

Reply:
Thanks yeah, used a `wait_for_id` which will be less flaky. So will tell us that reports have started. Really this test is checking that a report running with a long timeout won't get killed to running the minimal report does nothing for us here. I will remove that to make test simpler